### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by TdrTs**

### CODE REVIEW

File: contract.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    printf("b) Go back\n");
+    char choice;
+    choice = getchar();
+    if(choice == 'a') {
+        *confirmation = 1;
+    } else {
+        *state = 4;
+    }
+    getchar();
+}
```
Comment (Ranapop) [inline contract.c]: this function does several things that don't naturally go together:
* display cutlery
* display additional info 
* display payment amount
* display order options
* sign order
I would say displaying the cuttlery, add info, and payment amount would more naturally go with displaying the other data that is part of the order (food, drinks) --- maybe a displayOrder function?
You could keep the order options and signing in this function but you would have to rename it

File: contract.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+#ifndef FOOD_ORDERING2_CONTRACT_H
+#define FOOD_ORDERING2_CONTRACT_H
+
+#define FOOD_TYPE 20
+#define FOOD_NAME 30
+#define DRINK_NAME 10
+
+void finalContract(char user[20], char foodTypes[FOOD_NAME], double prices);
```
Comment (Ranapop) [inline contract.h]: function names are usually verbs not nouns

File: main.c [hunk 3]
```diff
@@ -1,174 +1,73 @@
...
-    int confirmation =0;
-    char foods[][20] = {"Pizza","Pasta","Salad"};
-    int noFoods[] = {3,2,4};
-    int noDrinks = 4;
-    int ok = 1;
-    char foodTypes[3][4][30] = {
+int main() {
+    printf("Welcome to Food Thingies! \nPlease sign in to continue!\n");
+    char user[20], pass[20], info[100];
+    int confirmation = 0, noDrinks = 4, ok = 1, noOfFoods = 3, choice, foodChoice, foodType, drinkChoice, cutleryChoice, state = 0, noFoods[] = {3,2,4};
```
Comment (Ranapop) [inline main.c]: such a long line is not acceptable, please properly format

File: main.c [hunk 4]
```diff
@@ -1,174 +1,73 @@
...
+    char foodTypes[3][4][FOOD_NAME] = {
             {"Pizza Carbonara", "Pizza Diavola", "Pizza Margherita"},
             {"Chicken alfredo", "Mac&cheese"},
             {"Tuna Salad",      "Chicken Salad", "Greek Salad", "Cobb"}
     };
-
-
-    double prices[3][4] = {
+    char drinkTypes[4][DRINK_NAME] = {"Coca-Cola", "Fanta", "Lipton", "Water"}, cutleryOptions[2][5] = {"Yes", "No"}, foods[][FOOD_TYPE] = {"Pizza","Pasta","Salad"};
```
Comment (Ranapop) [inline main.c]: such a long line is not acceptable, please properly format

File: main.c [hunk 5]
```diff
@@ -1,174 +1,73 @@
...
+    double drinkPrices[4] = {5,5,5,4}, totalPrice = 0, prices[3][4] = {
```
Comment (Ranapop) [inline main.c]: please properly format

File: userdata.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    choice = getchar();
+    getchar();
+    if(choice == 'a'+index) {
+        (*state)--;
+    } else {
+        (*state)++;
+    }
+
+    return  (choice - 'a');
+}
```
Comment (Ranapop) [inline userdata.c]: why getChoice in userdata?

File: contract.c [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    printf("Name: %s \n",user);
+    printf("Food items: \n");
+    printf("-- %s : %.f \n", foodTypes, prices);
+}
+
+int getTotalPrice(double foodPrice, int drinkChoice, char drinkType[], double drinkPrice){
+
+    int total = foodPrice;
+
+    if (drinkChoice < 4) {
```
Comment (Ranapop) [inline contract.c]: what's with this if? what is 4? ah, magic numbers...

File: userdata.c (hunk 6 above)
Comment (TdrTs) [reply userdata.c]: I considered this input as a data provided by the user.

**Pull request merged**